includes/bp-members/classes/class-bp-members-endpoints.php

```diff
@@ -0,0 +1,402 @@
+			'type'              => 'array',
+			'default'           => array(),
+			'sanitize_callback' => 'wp_parse_id_list',
+		);
+
+		$params['populate_extras'] = array(
+			'description'       => __( 'Fetch extra meta for each user such as their full name, if they are a friend of the logged in user, their last activity time.', 'buddypress' ),
+			'default'           => true,
+			'type'              => 'boolean',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		return $params;
+	}
+
+	public function prepare_item_for_response( $activity, $request, $is_raw = false ) {
+		$data = array(
+			'author'                => $activity->user_id,
+			'component'             => $activity->component,
+			'content'               => $activity->content,
+			'date'                  => $this->prepare_date_response( $activity->date_recorded ),
+			'id'                    => $activity->id,
+			'link'                  => $activity->primary_link,
+			'parent'                => $activity->type === 'activity_comment' ? $activity->item_id : 0,
+			'prime_association'     => $activity->item_id,
+			'secondary_association' => $activity->secondary_item_id,
+			'status'                => $activity->is_spam ? 'spam' : 'published',
+			'title'                 => $activity->action,
+			'type'                  => $activity->type,
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+		$response->add_links( $this->prepare_links( $activity ) );
+
+		/**
+		 * Filter an activity value returned from the API.
+		 *
+		 * @param array           $response
+		 * @param WP_REST_Request $request Request used to generate the response.
+		 */
+		return apply_filters( 'rest_prepare_buddypress_activity_value', $response, $request );
+	}
+
+
+	/**
+	 * Retrieve activities.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param WP_REST_Request $request
+	 *
+	 * @return WP_REST_Request List of activity object data.
+	 */
+	public function get_items( $request ) {
+		$args = $this->_get_args($request);
+
+		$retval     = array();
+		if ( bp_has_members( $args ) ) :
+			while ( bp_members() ) : bp_the_member();
+				$member = [];
+				$member['user_id'] = bp_get_member_user_id();
+				$member['user_name'] = bp_get_profile_field_data( ['field'=>'Name', 'user_id'=>bp_get_member_user_id()] );
+				$member['user_nicename'] = bp_get_member_user_nicename();
+				$member['user_avatar'] = bp_get_member_avatar();
+				$member['user_permalink'] = bp_get_member_permalink();
+				$member['diary_title'] = bp_get_profile_field_data( ['field'=>'Diary Title', 'user_id'=>bp_get_member_user_id()] );
+				$member['total_friend_count'] = bp_get_member_total_friend_count();
```

Comment:
I'd sanitize all of these as well, eg `$member['user_id'] = intval( bp_get_member_user_id() );`